source/portable/NetworkInterface/NXP1060/NetworkInterface.c

```diff
@@ -0,0 +1,620 @@
+    ethernetifLocal->TxDataBuff    = &( txDataBuff_0[ 0 ] );
+    ethernetifLocal->base          = ( void * ) ENET_BASE;
+
+	/* Set the clock frequency. */
+	mdioHandle.resource.csrClock_Hz = EXAMPLE_CLOCK_FREQ;
+	mdioHandle.resource.base = ( void * ) ENET_BASE;
+
+	/* Set the configuration to auto-negotiate; set the phy to full duplex mode; the phy's address; and the speed. */
+	xConfig.autoNeg = pdTRUE;
+	xConfig.duplex = kPHY_FullDuplex;
+	xConfig.phyAddr = BOARD_ENET0_PHY_ADDRESS;
+	xConfig.speed = kPHY_Speed100M;
+
+	configPRINTF( ( "Starting PHY initialization.\r\n" ) );
+
+	xStatus = PHY_Init( &phyHandle, &xConfig );
+
+	if( xStatus != kStatus_Success )
+	{
+		xResult = pdFAIL;
+		configPRINTF( ( "Failed to initialize the PHY..." ) );
+	}
+	else
+	{
+		configPRINTF( ( "Configuration successful. Waiting for autonegotiation to complete...\r\n" ) );
+
+		do
+		{
+			readStatus = PHY_GetAutoNegotiationStatus( &phyHandle, &autoNegotiationComplete );
+
+			if( autoNegotiationComplete == true )
+			{
+				break;
+			}
+
+			/* Try for only a limited number of times. */
+			if( ucCounter++ > MAX_AUTONEG_FAILURE_COUNT )
+			{
+				break;
+				xResult = pdFAIL;
+			}
+
+			vTaskDelay( pdMS_TO_TICKS( 10 ) );
+		}
+		while( readStatus == kStatus_Success );
+	}
+
+
+	if( xResult == pdPASS )
+	{
+		if( autoNegotiationComplete == false )
+		{
+			configPRINTF( ( "Failed to complete auto-negotiation. Using default values." ) );
+
+			xConfig.autoNeg = pdFALSE;
+
+			xStatus = PHY_Init( &phyHandle, &xConfig );
+			if( xStatus != kStatus_Success )
+			{
+				xResult = pdFAIL;
+				configPRINTF( ( "Failed to re-initialize the PHY..." ) );
+			}
+			else
+			{
+				configPRINTF( ( "PHY re-initialization successful." ) );
+			}
+		}
+		else
+		{
+			configPRINTF( ( "Auto-negotiation complete.\r\n" ) );
+		}
+
+		ucCounter = 0;
+		do
+		{
+			readStatus = PHY_GetLinkStatus( &phyHandle, &LinkUp );
+
+			if( LinkUp == true )
+			{
+				break;
+			}
+
+			vTaskDelay( pdMS_TO_TICKS( 1000 ) );
+
+			/* Try for only a limited number of times. */
+			if( ucCounter++ > MAX_AUTONEG_FAILURE_COUNT )
+			{
+				break;
+				xResult = pdFAIL;
+			}
+		}
+		while( readStatus == kStatus_Success );
+
+		if( LinkUp == false )
+		{
+			xResult = pdFAIL;
+			configPRINTF( ( "Failed to get the link up." ) );
+		}
+		else
+		{
+			/* Success in auto-negotiation and the link is up. */
+			configPRINTF( ( "Link up." ) );
+		}
+	}
+
+	if( xResult != pdFAIL )
+	{
+		readStatus = PHY_GetLinkSpeedDuplex( &phyHandle, &speed, &duplex );
+		if( readStatus != kStatus_Success )
+		{
+			xResult = pdFAIL;
+		}
+	}
+
+	if( xResult != pdFAIL )
+	{
+	    /* We were able to get the link up. Set up all other required things - such as DMA,
+		 * network output etc. */
+
+		/* The handler task is created at the highest possible priority to
+		 * ensure the interrupt handler can return directly to it. */
+		xResult = xTaskCreate( prvEMACHandlerTask,
```

Comment:
The program might get here more than one time and create a new instance of the task, don't you think?
I would recommend to postpone creating the task until the moment that the link-status is high, and all has been initialise. If not, the task starts running (at a higher priority than the IP-task) and work with the EMAC.